chain/block.go

```diff
@@ -708,40 +728,95 @@ func (b *StatelessBlock) Processed() bool {
 }
 
 // View returns the [merkledb.TrieView] of the block (representing the state
-// post-execution) or returns the accepted state if the block is accepted or is height 0.
+// post-execution) or returns the accepted state if the block is accepted or
+// is height 0 (genesis).
 //
-// If [b.view] is nil, this function will either return an error or will
-// run verification depending on the value of [verify].
+// If [b.view] is nil (not processed), this function will either return an error or will
+// run verification depending on the value of [blockRoot].
 //
 // Invariant: [View] with [verify] == true should not be called concurrently, otherwise,
 // it will result in undefined behavior.
-func (b *StatelessBlock) View(ctx context.Context, verify bool) (state.View, error) {
+func (b *StatelessBlock) View(ctx context.Context, blockRoot *ids.ID) (state.View, error) {
 	ctx, span := b.vm.Tracer().Start(ctx, "StatelessBlock.View",
 		oteltrace.WithAttributes(
-			attribute.Bool("verify", verify),
+			attribute.Stringer("blockRoot", blockRoot),
```

Comment:
Does this work? The stringer for `ids.ID` takes in an `ids.ID` - not an `*ids.ID`... so I feel like `blockRoot = nil` should cause this to panic... no?

Reply:
I have no idea.